### ISSUE TITLE
[Request] Add setting for dialogue style

### ISSUE THREAD

**Issue by Flitskikker** [SubtitleEdit/subtitleedit#3416]
In the subtitling world, every country has their own guidelines and style. One of the elements that are different across countries, is the dialogue style. At the moment, Subtitle Edit only supports one dialogue style. I think it would be useful to allow the user to configure the dialogue style used.

As far as I can tell, these are the 4 most common dialogue styles:

_English fansubs / Scandinavian subtitles (currently the only supported style):_
```
- Lorem ipsum dolor sit amet.
- Consectetur adipiscing elit.
```

_English retail subtitles:_
```
-Lorem ipsum dolor sit amet.
-Consectetur adipiscing elit.
```

_Dutch professional subtitles:_
```
Lorem ipsum dolor sit amet.
-Consectetur adipiscing elit.
```

_Dutch fansubs / Flemish subtitles:_
```
Lorem ipsum dolor sit amet.
- Consectetur adipiscing elit.
```

However, there are countries that have dialog dashes mid-line, and there are countries that use the en dash (–) instead of a hyphen (-) as a dialogue dash, so I think it would be best to implement generic settings as follows:

![se3](https://user-images.githubusercontent.com/3516155/53378602-7726f500-3966-11e9-8049-35fe77a2c66c.png)

These settings would then be used across the whole application, for example:
- For the "Merge selected lines a dialogue" option
- When splitting a dialogue
- For detecting errors (e.g. ignore "Fix missing space" for retail style)
- If possible, an option to convert between styles

I'm following the professional Dutch style, so at the moment I do everything dialogue-related manually. It would be nice to finally be able to use these features. 😄

Thanks in advance!

**Comment by Ding-adong** [comment]
- denotes different speakers therefore having...
- blah speaker 1
- blah speaker 2
makes perfect sense. Perhaps the Dutch people should evolve to using better subtitling.

**Comment by Flitskikker** [comment]
In Dutch, we call it the "second speaker dash".
A single dash is also used in other languages than Dutch, such as Bulgarian, Croatian, Icelandic, Slovenian, Serbian... probably more.
Therefore, I think many users would benefit from such an option.

**Comment by darnn** [comment]
For what it's worth, in Hebrew it's usually the same as in the above example of Dutch professional subtitles.

**Comment by OmrSi** [comment]
In Arabic, we always use the first one, whether it is for Netflix, Amazon or fansubbing.

**Comment by marb99** [comment]
In portuguese European and Brazilian always use the first one, the same as OmrSi

**Comment by OmrSi** [comment]
marb99 I think it's the one most used...

**Comment by niksedk** [comment]
Dialog styles should now be available in latest beta:  https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.13/SubtitleEditBeta.zip

For more info see pull request #3993

Please test :)
Also, please check ´profiles´.

**Comment by Ding-adong** [comment]
Not working properly. When there is a dash before a single person speaking, the 'remove dialogue hyphen in single line' will not work. ie.

`- Hello.`

In order to work I had to uncheck 'Break long lines'. I think the Break long line should be at the bottom of the 'what to fix' list in fix common errors, so it only activates after all corrections had been sorted.

**Comment by niksedk** [comment]
@Ding-adong: Could you re-try in latest beta: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.13/SubtitleEditBeta.zip ?
If it still happens, a screenshot or two would be very nice (as I cannot re-create it now).

**Comment by Ding-adong** [comment]
Nope.

`- In the new company (line break)
that I'm starting up.`

Will not work unless i unbreak the line into one long line. My max length of single line is 34 and if 

`- In the new company.`

it will work. But Not when over max length of single line where there is a line break as shown in first example.

**Comment by niksedk** [comment]
@Ding-adong: thx for the example :)
Beta updated: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.13/SubtitleEditBeta.zip

**Comment by Ding-adong** [comment]
Congrats man.

That worked as intended. Will let you know if there is any new issue. So far none.

**Comment by Ding-adong** [comment]
Oh heck doesn't work properly.

Worked when last character on 1st line is letter.
```
- You know, that's okay
We're in kind of a hurry.
```

Worked when last character on 1st line is comma.
```
- You know, that's okay,
We're in kind of a hurry.
```

Failed when last character on 1st line is full stop.
```
- You know, that's okay.
We're in kind of a hurry.
```

Failed when last character on 1st line is ?
```
- You know, that's okay?
We're in kind of a hurry.
```

**Comment by OmrSi** [comment]
@niksedk So, I got to do some testing now on a subtitle.
This is an Arabic subtitle with no punctuation at the end of lines, it's from iTunes.
Check out line 891: [Knives.Out.2019.BluRay.zip](https://github.com/SubtitleEdit/subtitleedit/files/4229516/Knives.Out.2019.BluRay.zip)
There are two reasons for not detecting it here, the first one is that there is a Unicode control char at the beginning of the line and the second is that it doesn't end with punctuation.

Side note: Can you rename "Tools -> Split long lines..." to "Break long lines"?
That's what it does, it breaks, not splits.

**Comment by Ding-adong** [comment]
There was some changes in how fix (add dash) and fix (remove dash), fix common errors, works last year. In the last few months I had noticed a single speaker has a dash per line, making the assumption there are two speakers. I had to correct it while watching the programme. It happens when there are two sentences, broken into two lines like below. Today I just found out why.

 ```
- So what is that?
What does that mean?
```

Fix (add dash) change it into
 ```
- So what is that?
- What does that mean?
```
when it is suppose to be
 ```
So what is that?
What does that mean?
```

This is wrong. The purpose of fix (add dash) is to fix 'second speaker dash' such as dutch subs into dash for each speaker in eng sub. See top of thread for examples.

 ```
So what is that?
- What does that mean?
```

becomes
 ```
- So what is that?
- What does that mean?
```
is correct.

**Comment by Ding-adong** [comment]
This is an example why 'break long lines' in fix common errors need to done after 'fix (remove dash)'.

```
you would've been okay?
- Then I would still have it, yeah. (35 characters which is more than max chars per lines of 34)
```
becomes
```
you would've been okay? - Then
I would still have it, yeah.
```
then 'fix dialogs on one line' becomes
```
you would've been okay?
- Then I would still have it, yeah.
```
back to the same as the first. A horrible loop that refused to use 'fix (add dash)' until I uncheck 'break long lines.'

It should work like this.
```
you would've been okay?
- Then I would still have it, yeah. (35 characters which is more than max chars per lines of 34)
```
becomes (fix (add dash))
```
- you would've been okay?
- Then I would still have it, yeah. (35 characters which is more than max chars per lines of 34)
```
becomes (break long lines)
```
1; start time; end time; duration; You would've been okay?
2; start time; end time; duration; Then I would still have it, yeah.
```

**Comment by niksedk** [comment]
Closed via #3993
